calico/felix/ipsets.py

```diff
@@ -320,22 +316,80 @@ def _on_endpoint_data_update(self, endpoint_id, endpoint_data):
         # from one profile to another but keeps the same tag.
         for profile_id, tag in added_tags:
             for ip in new_ips:
-                self._add_mapping(tag, profile_id,  endpoint_id, ip)
+                self.tag_membership_index.add_mapping(tag, profile_id,
```

Comment:
Sometimes you save off the add_mapping / remove_mapping to a new variable, and sometimes you don't. Would read better to be consistent (and should they be class variables to make the whole thing even simpler?).